frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.unit.spec.tsx

```diff
@@ -244,95 +244,131 @@ describe("actions > containers > ActionCreator > FormCreator", () => {
+            });
+
+            setup({
+              parameters: [parameter],
+              formSettings: {
+                type: "form",
+                fields: {
+                  [parameter.id]: fieldSettings,
+                },
+              },
+              isPublic,
+            });
+
+            expect(
+              screen.getByTestId("action-warning-banner"),
+            ).toBeInTheDocument();
+          });
+
+          it("shows a warning banner for implicit action", () => {
+            const parameter = makeParameter({ required: true });
+            const fieldSettings = createMockImplicitActionFieldSettings({
+              id: parameter.id,
+              required: true,
+              defaultValue: undefined,
+              hidden: true,
+            });
+
+            setup({
+              parameters: [parameter],
+              formSettings: {
+                type: "form",
+                fields: {
+                  [parameter.id]: fieldSettings,
+                },
+              },
+              isPublic,
+            });
+
+            expect(
+              screen.getByTestId("action-warning-banner"),
+            ).toBeInTheDocument();
+          });
+        });
+      },
+    );
 
-    it("doesn't show a warning banner if no required params hidden", () => {
-      const parameter = makeParameter({ required: true });
-      const fieldSettings = createMockImplicitActionFieldSettings({
-        id: parameter.id,
-        hidden: false,
-      });
-
-      setup({
-        parameters: [parameter],
-        formSettings: {
-          type: "form",
-          fields: {
-            [parameter.id]: fieldSettings,
+    describe.each([{ isPublic: true }, { isPublic: false }])(
+      "when public: $isPublic",
+      ({ isPublic }) => {
+        describe.each([
+          { required: false, hidden: false, hasDefaultValue: false },
+          { required: false, hidden: false, hasDefaultValue: true },
+          { required: false, hidden: true, hasDefaultValue: false },
+          { required: false, hidden: true, hasDefaultValue: true },
+          { required: true, hidden: false, hasDefaultValue: false },
+          { required: true, hidden: false, hasDefaultValue: true },
+          { required: true, hidden: true, hasDefaultValue: true },
+        ])(
+          `when there is no hidden, required parameter without default value`,
+          ({ required, hidden, hasDefaultValue }) => {
+            it("does not show a warning banner for query action", () => {
+              const defaultValue = "foo bar";
+              const parameter = makeParameter({ required });
+              const fieldSettings = makeFieldSettings({
+                id: parameter.id,
+                inputType: "string",
+                required,
+                defaultValue: hasDefaultValue ? defaultValue : undefined,
+                hidden,
+              });
+
+              setup({
+                parameters: [parameter],
```

Comment:
All test cases here always use 1 field/parameter (as if there was an assumption that `Array.prototype.some` is correctly called under the hood).
I'm not sure if we need to go into such detail but do you think it would make sense to add tests for cases when there's 0 and/or more than 1 field/parameter?

Reply:
good point

Reply:
added a second parameter and also a special case with no parameters passed